paywall/src/utils/polling.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * poll at a regular interval, with the ability to pause polling or end it
+ */
+export default function pollWithConditions(
+  pollingCallback,
+  intervalMs,
+  disablePolling,
+  handleError = () => {}
+) {
+  async function poll() {
+    try {
+      if (disablePolling()) return
```

Comment:
Why use a function for this? And if we need to, why assume that this will always be sync?